tests/ut/test_decorators.py

```diff
@@ -590,6 +590,13 @@ async def bar():
 
         assert foo.cache != bar.cache
 
+    async def test_key_builder(self):
+        @multi_cached("keys", key_builder=lambda key, _, keys: key + 1)
+        async def incr(keys=None):
+            return {k: k + 1 for k in keys}
+        
+        assert incr(keys=[1]) == {1: 3}
```

Comment:
```suggestion
        @multi_cached("keys", key_builder=lambda key, _, keys: key + 1)
        async def incr(keys=None):
            return {k: k * 3 for k in keys}
        
        assert incr(keys=(1,)) == {1: 3}
        cached_value = await incr.cache.get(2)
        assert cached_value == 3
        assert not await incr.cache.exists(1)
```

Maybe something like this is more comprehensive.